test/src/System/Time.cpp

```diff
@@ -0,0 +1,172 @@
+        }
+
+        SECTION("operator<")
+        {
+            CHECK(sf::seconds(54.3) < sf::seconds(55));
+            CHECK(sf::microseconds(10) < sf::milliseconds(10));
+        }
+
+        SECTION("operator>")
+        {
+            CHECK(sf::milliseconds(100'000) > sf::seconds(55));
+            CHECK(sf::microseconds(1) > sf::seconds(0.0000001));
+        }
+
+        SECTION("operator<=")
+        {
+            CHECK(sf::milliseconds(100) <= sf::milliseconds(100));
+            CHECK(sf::seconds(0.0012) <= sf::microseconds(1'201));
+        }
+
+        SECTION("operator>=")
+        {
+            CHECK(sf::milliseconds(100) >= sf::milliseconds(-100));
+            CHECK(sf::microseconds(1'201) >= sf::seconds(0.0012));
+        }
+
+        SECTION("operator-")
+        {
+            CHECK(sf::seconds(-1.0) == -sf::seconds(1.0));
+            CHECK(sf::microseconds(1'234) == -sf::microseconds(-1'234));
+        }
+
+        SECTION("operator+")
+        {
+            CHECK(sf::seconds(1.0) + sf::seconds(1.0) == sf::seconds(2.0));
+            CHECK(sf::milliseconds(400) + sf::microseconds(400) == sf::microseconds(400400));
+        }
+
+        SECTION("operator+=")
+        {
+            sf::Time time = sf::seconds(1.5);
+            time += sf::seconds(1.0);
+            CHECK(time == sf::seconds(2.5));
+        }
+
+        SECTION("operator-")
+        {
+            CHECK(sf::seconds(1.0) - sf::seconds(1.0) == sf::seconds(0.0));
+            CHECK(sf::milliseconds(400) - sf::microseconds(400) == sf::microseconds(399600));
+        }
+
+        SECTION("operator-=")
+        {
+            sf::Time time = sf::seconds(1.5);
+            time -= sf::seconds(10.0);
+            CHECK(time == sf::seconds(-8.5));
+        }
+
+        SECTION("operator*")
+        {
+            CHECK(sf::seconds(1.0) * 2.0f == sf::seconds(2.0));
+            CHECK(sf::seconds(12.0) * 0.5f == sf::seconds(6.0));
+            CHECK(sf::seconds(1.0) * sf::Int64(2) == sf::seconds(2.0));
+            CHECK(sf::seconds(42.0) * sf::Int64(2) == sf::seconds(84.0));
+            CHECK(2.0f * sf::seconds(1.0) == sf::seconds(2.0));
+            CHECK(0.5f * sf::seconds(12.0) == sf::seconds(6.0));
+            CHECK(sf::Int64(2) * sf::seconds(1.0) == sf::seconds(2.0));
+            CHECK(sf::Int64(2) * sf::seconds(42.0) == sf::seconds(84.0));
+        }
+
+        SECTION("operator*=")
+        {
+            sf::Time time = sf::milliseconds(420);
+            time *= 0.1f;
+            CHECK(time == sf::milliseconds(42));
+            time *= sf::Int64(10);
+            CHECK(time == sf::milliseconds(420));
+        }
+
+        SECTION("operator/")
+        {
+            CHECK(sf::seconds(1.0) / 2.0f == sf::seconds(0.5));
+            CHECK(sf::seconds(12.0) / 0.5f == sf::seconds(24.0));
+            CHECK(sf::seconds(1.0) / sf::Int64(2) == sf::seconds(0.5));
+            CHECK(sf::seconds(42.0) / sf::Int64(2) == sf::seconds(21.0));
```

Comment:
When testing floating point operations, I read it is generally a bad idea to rely on the == operator, even though it might be `true` on some architecture, these tests might fail on others. Rather do a comparison towards a "minimum tolerable error" with an implementation-specific "epsilon" like defined in `std::numeric_limits<float>::epsilon`.

Reply:
Totally agree with phgroe's comment. To add, both Catch2 and doctest provide built-in assertions for such checks:
* [Catch2](https://github.com/catchorg/Catch2/blob/devel/docs/assertions.md#floating-point-comparisons)
* [doctest](https://github.com/doctest/doctest/blob/master/doc/markdown/assertions.md#floating-point-comparisons)

Reply:
Relying on perfect floating point equality can be a problem when doing calculations with floating point numbers but under the hood all these times are represented by a signed integer. Unless tests start failing on some other platform I'd rather not dull these tests with pseudo-equalities since those fuzzy comparisons are less rigorous than demanding perfect equality.

Reply:
> Those might fail some (future?) function-style cast warnings. Maybe use static_cast even if it's ugly.
 
 Would these count as function-style casts since `sf::seconds`, `sf::milliseconds`, and `sf::microseconds` are not constructors? The function accepts a float so passing a double is sloppy and would be caught by `-Wconversion` so maybe I ought to only pass floats as arguments but I don't think we need to add any `static_cast`s to this file.

Reply:
> Those might fail some (future?) function-style cast warnings. Maybe use static_cast even if it's ugly.
 
Good point. I fixed that. I've noticed how SFML is very strict about not using c-style casts. If we're not already using `-Wold-style-cast` we should add that to our set of compiler warnings.

Reply:
```cpp
bool operator ==(Time left, Time right)
{
    return left.asMicroseconds() == right.asMicroseconds();
}
```
Comparing times is implemented as comparing integers so as long as it's implemented this way, I think we can rely on perfect equality at all times. If the implementation changes and becomes less precise, the tests should catch that (if my tests are rigorous enough).
```cpp
Time seconds(float amount)
{
    return Time(static_cast<Int64>(amount * 1000000));
}
```
Floating point numbers enter the picture with `sf::seconds` but they're immediately cast to the underlying type, `sf::Int64`. I'm working on some additional assertions that tests how `sf::seconds` works since something like `sf::seconds(1'000'000.0f)` does not equal 1,000,000,000,000 microseconds. In fact, there's relatively little we can do to ensure that a specific floating point literal corresponds to a specific number of microseconds. In these cases, you're absolutely right that we need to use fuzzy comparisons to correctly test the platform-specific nature of floating point numbers. I'm not exactly sure how to test something like 1 million seconds, but using an epsilon of some value does make a lot of sense there.

Reply:
I had to add `Approx(x).margin(1)` matchers to a few assertions where the weirdness of floating point numbers were changing the resulting microseconds value by 1. While these tests don't ever directly compare floating point numbers, they do compare the results of floating point calculations so similar reasoning applies. I tried to avoid using that approximate matcher as much as possible since for the most part it wasn't necessary.